### PR TITLE
MPDX-8425 Require Task Action on Mass Actions

### PR DESCRIPTION
**Pull request by caleballdrin**
## Description
https://jira.cru.org/secure/RapidBoard.jspa?rapidView=3&view=detail&selectedIssue=MPDX-8425&sprint=1282#
- Require Task Action on mass edit modal

## Checklist:

- [ ] I have given my PR a title with the format "MPDX-(JIRA#) (summary sentence max 80 chars)"
- [ ] I have applied the appropriate labels. (_Add the label "On Staging" to get the branch automatically merged into staging._)
- [ ] I have requested a review from another person on the project

### CODE REVIEW

File: src/utils/functions/getLocalizedResultStrings.ts [hunk 1]
```diff
@@ -37,7 +37,7 @@ export const getLocalizedResultString = (
       return t('No Response Yet');
 
     case DisplayResultEnum.AppointmentResultNotInterested:
-      return t('Does not want to meet');
+      return t('Not Interested');
```
Comment (dr-bizz) [inline src/utils/functions/getLocalizedResultStrings.ts:40]: Good catch

Comment (dr-bizz) [reply src/utils/functions/getLocalizedResultStrings.ts:40]: Was this missed when we did Task Phases?

File: src/components/Task/MassActions/EditTasks/MassActionsEditTasksModal.tsx [hunk 2]
```diff
@@ -185,6 +201,11 @@ export const MassActionsEditTasksModal: React.FC<
                     }
                     // None and null are distinct values: null leaves the action unchanged and None changes the action to None
                     preserveNone
+                    required={!!taskPhase}
```
Comment (dr-bizz) [inline src/components/Task/MassActions/EditTasks/MassActionsEditTasksModal.tsx:204]: It seems it is auto-selecting options when it opens the activity and then you click away.

File: src/components/Task/MassActions/EditTasks/MassActionsEditTasksModal.test.tsx [hunk 3]
```diff
@@ -86,47 +86,80 @@ describe('MassActionsEditTasksModal', () => {
...
+        </ThemeProvider>
+      </LocalizationProvider>,
+    );
+    const taskTypeField = await findByRole('combobox', { name: 'Task Type' });
+    const actionField = await findByRole('combobox', { name: 'Action' });
+
+    userEvent.click(taskTypeField);
+    userEvent.click(await findByRole('option', { name: 'Appointment' }));
+
+    await waitFor(() => expect(getByText('Save')).toBeDisabled());
```
Comment (dr-bizz) [inline src/components/Task/MassActions/EditTasks/MassActionsEditTasksModal.test.tsx:156]: For bonus points, you should make the error message show in the test

**Review by dr-bizz — CHANGES_REQUESTED** [review]
The code is looking good. I found an error when you select a phase, it shows the activity dropdown, but if you click away from the dropdown, it autoselects the first option, which is the first issue, and the second issue is that it doesn't recognise it is selected.

If you select another option or select it properly, it works. I know these errors were here before, but since we are making the activity dropdown required if the task phase is selected, it means we will have to fix them.

Screenshot 2024-11-04 at 1 37 20 PM

File: src/utils/functions/getLocalizedResultStrings.ts (hunk 1 above)
Comment (caleballdrin) [reply src/utils/functions/getLocalizedResultStrings.ts:40]: Yeah, we missed it. In the future, we should get it from the API.

File: src/components/Task/MassActions/EditTasks/MassActionsEditTasksModal.test.tsx (hunk 3 above)
Comment (caleballdrin) [reply src/components/Task/MassActions/EditTasks/MassActionsEditTasksModal.test.tsx:156]: I do love bonus points, but I spent 30 mins and couldn't get it to find the text... I think as long as it is disabled we can know it is working correctly.

**Review by dr-bizz — CHANGES_REQUESTED** [review]
It still does it somewhat. If you get to the activity and then press the downward arrow, then click on another field, it selects the item you were on, then doesn't allow the user to continue.
Screenshot 2024-11-04 at 4 35 09 PM

**Review by dr-bizz — APPROVED** [review]
Awesome work! Thank you for fixing this!

**Pull request merged**